src/envoy/auth/pubkey_cache.h

```diff
@@ -0,0 +1,102 @@
+/* Copyright 2017 Istio Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+#ifndef AUTH_PUBKEY_CACHE_H
+#define AUTH_PUBKEY_CACHE_H
+
+#include <chrono>
+#include <unordered_map>
+
+#include "config.h"
+
+namespace Envoy {
+namespace Http {
+namespace Auth {
+
+// Struct to hold an issuer cache item.
+class PubkeyCacheItem {
+ public:
+  PubkeyCacheItem(const IssuerInfo& config) : config_(config) {
+    if (!config_.pubkey_value.empty()) {
+      SetKey(config_.pubkey_value);
+    }
+  }
+
+  // Return true if cached pubkey is expired.
+  bool Expired() const {
+    return config_.pubkey_cache_expiration_sec > 0 &&
+           std::chrono::system_clock::now() >= expiration_time_;
```

Comment:
use `steady_clock` to manage expirations of cache.

Reply:
Done